dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/ndk/DatadogNdkCrashHandler.kt

```diff
@@ -0,0 +1,224 @@
+            if (crashLogFiles != null && crashLogFiles.isNotEmpty()) {
+                readFromFile(
+                    crashLogFiles[0],
+                    NdkCrashLog::class.java,
+                    "Malformed ndk crash error log",
+                    "Error while trying to read the ndk crash log"
+                )?.let {
+                    handleNdkCrashLog(it, searchLastRumViewEvent())
+                }
+            }
+            clearCrashLog()
+        }
+    }
+
+    private fun searchLastRumViewEvent(): RumEvent? {
+        val lastViewEventFiles =
+            ndkCrashDataDirectory.listFiles { _, name ->
+                name == RumFileWriter.LAST_VIEW_EVENT_FILE_NAME
+            }
+        if (lastViewEventFiles != null && lastViewEventFiles.isNotEmpty()) {
+            return readFromFile(
+                lastViewEventFiles[0],
+                RumEvent::class.java,
+                "Malformed RUM ViewEvent log",
+                "Error while trying to read the last rum view event log"
+            )
+        }
+        return null
+    }
+
+    private fun <T> readFromFile(
+        file: File,
+        type: Class<T>,
+        parsingErrorMessage: String,
+        ioErrorMessage: String
+    ): T? {
+        return try {
+            val serializedData = file.readText(Charsets.UTF_8)
+            @Suppress("UNCHECKED_CAST")
+            if (type == RumEvent::class.java) {
+                rumEventDeserializer.deserialize(serializedData) as? T
+            } else {
+                fromJson(serializedData) as T
+            }
+        } catch (e: JsonSyntaxException) {
+            sdkLogger.e(parsingErrorMessage, e)
+            null
+        } catch (e: IOException) {
+            sdkLogger.e(ioErrorMessage, e)
+            null
+        }
+    }
+
+    private fun handleNdkCrashLog(ndkCrashLog: NdkCrashLog, lastRumViewEvent: RumEvent?) {
+        val errorLogMessage = NDK_ERROR_LOG_MESSAGE.format(ndkCrashLog.signalName)
+        val bundledViewEvent = lastRumViewEvent?.event as? ViewEvent
+        val logAttributes: Map<String, String>
+        if (lastRumViewEvent != null && bundledViewEvent != null) {
+            logAttributes = mapOf(
+                LogAttributes.RUM_SESSION_ID to bundledViewEvent.session.id,
+                LogAttributes.RUM_APPLICATION_ID to bundledViewEvent.application.id,
+                LogAttributes.RUM_VIEW_ID to bundledViewEvent.view.id,
+                LogAttributes.ERROR_STACK to ndkCrashLog.stacktrace
+            )
+            // update the error count
+            val toSendRumEvent = resolveFromLastRumViewEvent(lastRumViewEvent, bundledViewEvent)
+            val toSendErrorEvent = resolveFromLastRumViewEvent(
+                errorLogMessage,
+                ndkCrashLog,
+                lastRumViewEvent,
+                bundledViewEvent
+            )
+            asyncRumWriter.write(toSendRumEvent)
+            asyncRumWriter.write(toSendErrorEvent)
```

Comment:
What about the `< 4h` rule discussed in the RFC? Sending RUM View event for session older than `4h` may lead to BE reducer's inconsistency.

Reply:
Yup...I was just reviewing the RFC now for making sure I am not missing anything in there.

Reply:
I also have to make sure the ViewEvent has `is_active=false`